### PR TITLE
RHIDP-2069 - [QE] Automate Scenario - Include HTTP Request as pre-loaded Plugin to RHDH

### PR DESCRIPTION
**Pull request by gustavolira**
## Description

[QE] Automate Scenario - Include HTTP Request as pre-loaded Plugin to RHDH

## Which issue(s) does this PR fix

[- Fixes #?]
https://issues.redhat.com/browse/RHIDP-2069

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: .ibm/pipelines/openshift-ci-tests.sh [hunk 1]
```diff
@@ -234,9 +234,14 @@ run_tests() {
 
   droute_send $release_name $project
 
-  exit ${RESULT}
+  echo ${RESULT}
+
+  if [ ${RESULT} -ne 0 ]; then
+    exit ${RESULT}
```
Comment (subhashkhileri) [inline .ibm/pipelines/openshift-ci-tests.sh:240]: In this case, if the showcase project fails, the RBAC test will not be executed as it will simply exit.

**Review by subhashkhileri — CHANGES_REQUESTED** [review]

File: .ibm/pipelines/openshift-ci-tests.sh (hunk 1 above)
Comment (gustavolira) [reply .ibm/pipelines/openshift-ci-tests.sh:240]: @subhashkhileri do you have a suggestion ?

File: .ibm/pipelines/openshift-ci-tests.sh [hunk 2]
```diff
@@ -234,9 +234,14 @@ run_tests() {
...
+  fi
```
Comment (subhashkhileri) [inline .ibm/pipelines/openshift-ci-tests.sh:241]: @gustavolira sure,

At the top of `openshift-ci-tests.sh` define
`OVERALL_RESULT=0`
```suggestion
  echo "$project RESULT: $RESULT"

  if [ ${RESULT} -ne 0 ]; then
    OVERALL_RESULT=1
  fi
```

then at the end of main() you can exit with: 
```
 initiate_deployments
 check_and_test ${RELEASE_NAME} ${NAME_SPACE}
 check_and_test ${RELEASE_NAME_RBAC} ${NAME_SPACE_RBAC}
 exit $OVERALL_RESULT
```

also update the `check_and_test()` as : 
```
check_and_test() {
  local release_name=$1
  local namespace=$2
  check_backstage_running $release_name $namespace
  backstage_status=$?
  echo "Display pods for verification..."
  oc get pods -n $namespace
  if [ $backstage_status -ne 0 ]; then
    echo "Backstage is not running. Exiting..."
    OVERALL_RESULT=1
  fi
  run_tests $release_name $namespace
}
```

**Review by subhashkhileri — CHANGES_REQUESTED** [review]

File: .ibm/pipelines/openshift-ci-tests.sh [hunk 3]
```diff
@@ -235,9 +236,13 @@ run_tests() {
 
   droute_send $release_name $project
 
-  exit ${RESULT}
+  echo "$project RESULT: $RESULT"
+    if [ "${RESULT}" -ne 0 ]; then
```
Comment (subhashkhileri) [inline .ibm/pipelines/openshift-ci-tests.sh:240]: It looks like there's an indentation mismatch.

**Pull request merged**